libs/category/testing/src/factories/filters/filter.factory.ts

```diff
@@ -20,10 +20,10 @@ export class DaffCategoryFilterFactory extends DaffModelFactory<DaffCategoryFilt
     super(<any>MockDaffCategoryFilter);
   }
 
-  create(partial?: Partial<DaffCategoryFilter>): DaffCategoryFilter {
+  create(partial: Partial<DaffCategoryFilter> = {}): DaffCategoryFilter {
     return {
       ...new this.type(),
-      ...faker.random.number({ min: 1, max: 2 }) === 2 ? this.equalFactory.create() : this.rangeFactory.create(),
```

Comment:
I had intentionally left this off as it doesn't make a lot of sense to provide a partial for a union type.

Reply:
There are still guaranteed overlaps in fields like `name` and `type`.

Reply:
Then we should modify the partial's type (and the internal factory calls) to match that more accurately.

Reply:
like this?

Reply:
like this?

Reply:
No, I think that's worse. I think if the user should be force to provide a "type" argument at least and we should use that type to infer the appropriate child factory to call.

Reply:
changed. lmk if you think I should test via a factory spy or if just asserting the type like I did is okay